### PR TITLE
Merge descriptor::TensorView into descriptor::Tensor

### PR DESCRIPTION
**Pull request by diyessi**
Merge TensorView and Tensor into Tensor
Leave behind alias and tensor_view accessors to ease transition

### CODE REVIEW

File: src/ngraph/descriptor/tensor.hpp [hunk 1]
```diff
@@ -16,55 +16,69 @@
...
-
     namespace descriptor
     {
-        class TensorView;
-        class Tensor;
-    }
-}
+        namespace layout
+        {
+            class TensorViewLayout;
```
Comment (rkimballn1) [inline src/ngraph/descriptor/tensor.hpp]: Is TensorViewLayout going to be renamed TensorLayout in a separate PR?

Comment (diyessi) [reply src/ngraph/descriptor/tensor.hpp]: It might merge in to Tensor as well since there's only one per tensor.

**Comment by shssf** [comment]
Have you tried to build `INTELGPU` backend with these changes?
